client.go

```
@@ -754,11 +738,15 @@ func (c *Client) CreateSession(ctx context.Context, cfg *uasc.SessionConfig) (*S
 		// Ensure we have a valid identity token that the server will accept before trying to activate a session
 		if c.cfg.session.UserIdentityToken == nil {
 			opt := AuthAnonymous()
-			opt(c.cfg)
+			if err := opt(c.cfg); err != nil {
```

revert

---

FYI: opt returning an error now, so here in that case that was me not the linter, but will do

---

OK, can we actually error here? Will this fail to create the session? If yes, then we should return the error.

---

If you want to add a `todo` then I suggest the following:

```go
// todo(sr): opt returns an error but we concluded that this call cannot 
// todo(sr): fail and that we do not want to stop creating the session
// todo(sr): hence we ignore it.
```

Don't just write `log.Error` since the next person don't know whether that is critical or not and if we do we spam other people's log files with stuff they cannot change.

---

I would suggest adding your suggested TODO. For the first call it might fail eventually once this TODO here gets fixed 
```
if !ok {
	// todo(fs): should we Fatal here?
	log.Printf("non-anonymous authentication already configured, ignoring")
	return nil
}
```
In case you decide to return the error there you must remember to handle the error in the call we talk about.